triedb/database.go

```diff
@@ -36,6 +36,8 @@ type Config struct {
 	IsVerkle  bool           // Flag whether the db is holding a verkle tree
 	HashDB    *hashdb.Config // Configs for hash-based scheme
 	PathDB    *pathdb.Config // Configs for experimental path-based scheme
+
+	DBOverride BackendConstructor // Injects an arbitrary backend implementation
```

Comment:
nit: IMO we can call this BackendConstructor as well (or a different name that refers to Backend)

Reply:
I agree that they should have the same name. What do you think of [these changes](https://github.com/ava-labs/libevm/pull/71)?